### PR TITLE
FEAT: Create ExtractDayOfYear operation and add its support to Clickhouse, CSV, MySQL, OmniSciDB, Pandas, Parquet, PostgreSQL, PySpark, SQLite and Spark

### PR DESCRIPTION
**Pull request by xmnlab**
Create ExtractDayOfYear operation and add its support to Clickhouse, CSV, MySQL, OmniSciDB, Pandas, Parquet, PostgreSQL, PySpark, SQLite and Spark

### CODE REVIEW

**Comment by xmnlab** [comment]
output from a local test:

```
(ibis-dev) ✔ ~/dev/quansight/ibis-project/ibis [create-op-day-of-year L|✚ 11] 
18:26 $ IBIS_TEST_MYSQL_PORT=3307 pytest -vv -x -k day_of_year ibis/tests/all/test_temporal.py::test_timestamp_extract
========================= test session starts =========================
platform linux -- Python 3.7.6, pytest-5.3.5, py-1.8.1, pluggy-0.13.0 -- /home/xmn/miniconda3/envs/ibis-dev/bin/python
cachedir: .pytest_cache
rootdir: /home/xmn/dev/quansight/ibis-project/ibis, inifile: setup.cfg
plugins: mock-2.0.0, cov-2.8.1, xdist-1.31.0, forked-1.1.2
collected 91 items / 78 deselected / 13 selected                      

ibis/tests/all/test_temporal.py::test_timestamp_extract[BigQuery-day_of_year] SKIPPED [  7%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[Clickhouse-day_of_year] PASSED [ 15%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[Csv-day_of_year] PASSED [ 23%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[HDF5-day_of_year] SKIPPED [ 30%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[Impala-day_of_year] XFAIL [ 38%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[MySQL-day_of_year] PASSED [ 46%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[OmniSciDB-day_of_year] PASSED [ 53%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[Pandas-day_of_year] PASSED [ 61%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[Parquet-day_of_year] PASSED [ 69%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[PostgreSQL-day_of_year] PASSED [ 76%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[PySpark-day_of_year] PASSED [ 84%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[SQLite-day_of_year] PASSED [ 92%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[Spark-day_of_year] PASSED [100%]

====== 10 passed, 2 skipped, 78 deselected, 1 xfailed in 24.97s =======

```

File: ci/docker-compose.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ services:
       - "sleep 80 && supervisord -c /etc/supervisord.conf -n"
 
   clickhouse:
-    image: yandex/clickhouse-server:18.12
+    image: yandex/clickhouse-server:18.14
```
Comment (xmnlab) [inline ci/docker-compose.yml:59]: depends on #2174

**Comment by xmnlab** [comment]
it is ready for review :)

**Comment by jreback** [comment]
looks fine, can you rebase and ping on green.

**Pull request merged**